components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/ClipboardSuggestionProvider.kt

```diff
@@ -54,6 +52,8 @@ class ClipboardSuggestionProvider(
         if ((requireEmptyText && text.isEmpty()) || !requireEmptyText) createClipboardSuggestion() else emptyList()
 
     private fun createClipboardSuggestion(): List<AwesomeBar.Suggestion> {
+        val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
```

Comment:
Isn't it an advantage that we don't have to do this when the user starts typing. This lookup will now happen `onInputChanged` where before this cost would've been hidden.

Reply:
I already tested this on the sample browser, in most cases, it's the same number of calls as the previous implementation. And the returned object is the same one every time. (in both implementations). But if you prefer the previous implementation (or decide it is the optimal one from a technical point of view) I can revert this commit.

Reply:
Personally, I would prefer this, because it will do less work when the user starts typing. To know for sure we'd need to measure it. So wouldn't make this particular change unless we know it's better.

Reply:
I dropped the commit.